app/controllers/trainees/subject_specialisms_controller.rb

```diff
@@ -64,7 +64,27 @@ def course_uuid
     end
 
     def subject_specialisms_for_position(position)
-      subject_specialisms[course_subject_attribute_name(position)]
+      specialisms = subject_specialisms[course_subject_attribute_name(position)]
+
+      if specialisms.blank? && first_course_is_modern_language? && position == 3
+        specialisms = subject_specialisms[course_subject_attribute_name(position - 1)]
+      end
```

Comment:
This is a bit messy. Suggestions welcome.

If the subjects are _Modern languages_ and _Biology_ say the first and second specialisms may be languages and therefore the biology specialism will need to be in the third position. However this doesn't align with the subject so we have to (potentially) look back at the previous position.